packages/loader/container-loader/README.md

```diff
@@ -198,3 +199,10 @@ This information can be used by a host to build appropriate UX that allows user
 Note that when an active connection is in place, it's just a matter of time before changes will be flushed to storage unless there is some source of continuous local changes being generated that prevents container from ever being fully saved. But if there is no active connection, because the user is offline, for example, then a document may stay in a dirty state for very long time.
 
 `Container.isDirty` can be used to get current state of container.
+
+## NoOps
+Definition: NoOps have MessageType.NoOp. "Empty operation message. Used to send an updated reference sequence number. Relay service is free to coalesce these messages or fully drop them, if another op was used to update Minimum Sequence Number to a number equal to or higher than referencedsequence number in Noop."
+
+At the end of a batch of ops, the client will send an NoOp to let the server know it's the end of the batch, almost like a benchmark? Sometimes we can also send noOp immediately, see both in CollabWindowTracker.scheduleSequenceNumberUpdate()
```

Comment:
I'd suggest not to describe particular behavior (that can change), but rather assumptions and expectations. I.e.
- noops can be coalesced by servvice. In practice this means that
   - noop can completely be swallowed (dropped) by service. In such case, there is, we observe a gap in clientSequenceNumber (otherwise all ops for a given clientId have sequential clientSequenceNumber values).
   - noop can be delayed by service (sequences later in time) and be reordered relative to ops that naturally follow it. That said, reordering can only happen across ops from different clients (if more ops are sent and sequenced by given client, then noop is simply dropped in such case).